src/Components/VaccinationDetails/AddVaccinationDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from "react";
+import TextFormField from "../Form/FormFields/TextFormField";
+import DateFormField from "../Form/FormFields/DateFormField";
+import ButtonV2 from "../Common/components/ButtonV2";
+import AutocompleteFormField from "../Form/FormFields/Autocomplete";
+import Spinner from "../Common/Spinner";
+
+const initForm: any = {
+  vaccination_center: "",
+  number_of_doses: "",
+  vaccine_name: "",
+  batch_number: "",
+  last_vaccinated_date: "",
+};
+export const AddVaccinationDetails = (props: any) => {
+  const [state, setState] = useState<any>({ ...initForm });
+  const handleFormFieldChange = (e: any) => {
+    setState((prev: any) => {
+      return {
+        ...prev,
+        [e.name]: e.value,
+      };
+    });
+  };
+  const hasError = props.disallowed.includes(state.vaccine_name);
+  return (
+    <div className=" flex flex-col gap-2">
```

Comment:
use form component and structure it similar to the existing forms, checkout patient register form for reference.